wprefs/bots/replace_except_new.py

```
@@ -0,0 +1,261 @@
+                group_regex = re.compile(r'\\(\d+)|\\g<(.+?)>')
+                last = 0
+                for group_match in group_regex.finditer(new):
+                    group_id = group_match.group(1) or group_match.group(2)
+                    with suppress(ValueError):
+                        group_id = int(group_id)
+
+                    try:
+                        replacement += new[last: group_match.start()]
+                        replacement += match.group(group_id) or ''
+                    except IndexError:
+                        raise IndexError(f'Invalid group reference: {group_id}\n Groups found: {match.groups()}')
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)